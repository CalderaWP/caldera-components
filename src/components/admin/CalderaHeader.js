import React from 'react';

/**
 * Creates the Caldera admin page header
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const CalderaHeader = (props) => {
	return(
		<div className="caldera-editor-header">
			<ul className="caldera-editor-header-nav">
				<li className="caldera-editor-logo">
					<span className="caldera-forms-name">
                        Caldera Forms: {props.name ? props.name : 'Caldera Forms'}
					</span>
				</li>
				{props.children}
			</ul>
		</div>
	);
};