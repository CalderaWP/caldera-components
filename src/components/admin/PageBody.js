import React from  'react';
/**
 * Wrapper for an admin page's content
 *
 * @since 1.7.0
 *
 * @param props
 * @returns {*}
 * @constructor
 */
export const PageBody = (props) => {
	return(
		<div
			style={{marginTop:'75px'}}
			className={'caldera-forms-admin-page-wrap'}
		>
			{props.children}
		</div>
	);
};