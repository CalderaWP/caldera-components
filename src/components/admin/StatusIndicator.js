import React from  'react';
import PropTypes from 'prop-types';

/**
 * A status indicator componet with CF Admin classes expected for CSS
 * @param props
 * @return {*}
 * @constructor
 */
export  const StatusIndicator = (props) =>{
	if( ! props.show){
		return null;
	}
	let className = 'cf-alert';
	if( props.success){
		className = `${className} cf-alert-success`;
	}else{
		className = `${className} cf-alert-error`;
	}

	return(
		<div className={'cf-alert-wrap'}>
			<p
				className={className}
			>
				{props.message}
			</p>
		</div>
	);

};
/**
 * Default props for the StatusIndicator compoents
 * @type {{message: shim, success: shim, show: shim}}
 */
StatusIndicator.propTypes = {
	message: PropTypes.string,
	success: PropTypes.bool,
	show: PropTypes.bool,
};