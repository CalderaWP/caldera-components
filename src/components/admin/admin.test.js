import renderer from 'react-test-renderer';
import React from  'react';

import {CalderaHeader} from './CalderaHeader';
import {PageBody} from './PageBody';
import {StatusIndicator} from './StatusIndicator';
describe( 'admin components', () => {
	it( 'CalderaHeader Matching Snapshot ', () => {
		const component = renderer.create(
			<CalderaHeader />
		);
		expect(component.toJSON()).toMatchSnapshot();

	});
	it( 'CalderaHeader Matching Snapshot with child li', () => {
		const component = renderer.create(
			<CalderaHeader

			>
				<li>Hi Roy</li>
			</CalderaHeader>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'PageBody Matching Snapshot ', () => {
		const component = renderer.create(
			<PageBody />
		);
		expect(component.toJSON()).toMatchSnapshot();

	});
	it( 'PageBody Matching Snapshot with child ', () => {
		const component = renderer.create(
			<PageBody

			>
				<div>Hi Roy</div>
			</PageBody>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it( 'StatusIndicator Matching Snapshot ', () => {
		const component = renderer.create(
			<StatusIndicator />
		);
		expect(component.toJSON()).toMatchSnapshot();

	});
	it( 'StatusIndicator Matching Snapshot with a message to show ', () => {
		const component = renderer.create(
			<StatusIndicator
				message={'Hi Roy'}
				success={true}
				show={true}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
	it( 'StatusIndicator Matching Snapshot with an error message to show ', () => {
		const component = renderer.create(
			<StatusIndicator
				message={'False Roy'}
				success={false}
				show={true}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
