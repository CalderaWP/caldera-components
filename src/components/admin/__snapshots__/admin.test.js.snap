// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin components CalderaHeader Matching Snapshot  1`] = `
<div
  className="caldera-editor-header"
>
  <ul
    className="caldera-editor-header-nav"
  >
    <li
      className="caldera-editor-logo"
    >
      <span
        className="caldera-forms-name"
      >
        Caldera Forms: 
        Caldera Forms
      </span>
    </li>
  </ul>
</div>
`;

exports[`admin components CalderaHeader Matching Snapshot with child li 1`] = `
<div
  className="caldera-editor-header"
>
  <ul
    className="caldera-editor-header-nav"
  >
    <li
      className="caldera-editor-logo"
    >
      <span
        className="caldera-forms-name"
      >
        Caldera Forms: 
        Caldera Forms
      </span>
    </li>
    <li>
      Hi Roy
    </li>
  </ul>
</div>
`;

exports[`admin components PageBody Matching Snapshot  1`] = `
<div
  className="caldera-forms-admin-page-wrap"
  style={
    Object {
      "marginTop": "75px",
    }
  }
/>
`;

exports[`admin components PageBody Matching Snapshot with child  1`] = `
<div
  className="caldera-forms-admin-page-wrap"
  style={
    Object {
      "marginTop": "75px",
    }
  }
>
  <div>
    Hi Roy
  </div>
</div>
`;

exports[`admin components StatusIndicator Matching Snapshot  1`] = `null`;

exports[`admin components StatusIndicator Matching Snapshot with a message to show  1`] = `
<div
  className="cf-alert-wrap"
>
  <p
    className="cf-alert cf-alert-success"
  >
    Hi Roy
  </p>
</div>
`;

exports[`admin components StatusIndicator Matching Snapshot with an error message to show  1`] = `
<div
  className="cf-alert-wrap"
>
  <p
    className="cf-alert cf-alert-error"
  >
    False Roy
  </p>
</div>
`;
